hexl/thread-pool/thread-pool-util.hpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <mutex>
+#include <thread>
+
+namespace intel {
+namespace hexl {
+
+// Enum for thread states
+enum class STATE {
```

Comment:
declaring the underlying type, I think will allow you to remove those static casts
```suggestion
enum class : int STATE {
```

Reply:
I tried it and the cast was still needed. But I can leave the type there.